siem_integrations/clx_query_service/clxquery/views.py

```diff
@@ -0,0 +1,61 @@
+import re
+import os
+import logging
+
+from clxquery import utils
+from clxquery.blazingsql_helper import BlazingSQLHelper
+from django.http import HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+
+
+log = logging.getLogger(__name__)
+
+file_path = os.environ.get("BLZ_READER_CONF")
+
+# Load tables configuration
+config = utils.load_yaml(file_path)
+configured_tables = set([table["table_name"] for table in config["tables"]])
+
+REGEX_PATTERN = r"main.([\w]+)"
+
+blz_helper = BlazingSQLHelper()
+
+
+@csrf_exempt
+def run_query(request, query):
+    if request.method == "GET":
+        # Check for the list of tables used in the query to prevent loading other tables into gpu memory
+        query_tables = set(re.findall(REGEX_PATTERN, query))
+        # Verify list of tables used in the query to make sure they are included in the configuration file
+        if query_tables.issubset(configured_tables):
+            try:
+                query_config = {}
+                query_config["tables"] = []
+                for table in config["tables"]:
+                    if table["table_name"] in query_tables:
+                        query_config["tables"].append(table)
+                query_config["sql"] = query
+                # Run query and get the results
+                df = blz_helper.run_query(query_config)
+                # Drop tables to free up memory
+                blz_helper.drop_table(query_tables)
+                # Convert cudf to pandas dataframe
+                df = df.to_pandas()
+                # Convert results to json format.
+                results = df.to_json(orient="records")
```

Comment:
Don't need to convert to pandas here. cudf has `to_json` method.

Reply:
I tried cudf `to_json` function but it doesn't have `return` and also this function uses pandas internally. Here is the sample reproducible code.

```
>>> import time
>>> from blazingsql import BlazingContext
>>> 
>>> config = {"tables":[{"table_name":"movies", "filepath":"/rapids/my_data/ml-25m/movies.csv"}]}
>>> 
>>> def run_query(query): 
...     bc = BlazingContext()
...     for table_config in config['tables']:
...         table_name = table_config['table_name']
...         file_path = table_config['filepath']
...         bc.create_table(table_name, file_path)
...     #Run query and get the results
...     df = bc.sql(query)
...     for table_config in config['tables']:
...         bc.drop_table(table_config['table_name'])
...         print("dropped table %s" %(table_config['table_name']))
...     return df
... 
>>> if __name__ == "__main__":
...     for i in range(0, 1):
...         df = run_query('SELECT * FROM main.movies LIMIT 10')
...         pd = df.to_pandas()
...         print(pd.to_json())
...         print('======')
...         print(df.to_json())
... 
listening: tcp://*:11541
BlazingContext ready
dropped table movies
{"0":{"0":"movieId","1":"1","2":"2","3":"3","4":"4","5":"5","6":"6","7":"7","8":"8","9":"9"},"1":{"0":"title","1":"Toy Story (1995)","2":"Jumanji (1995)","3":"Grumpier Old Men (1995)","4":"Waiting to Exhale (1995)","5":"Father of the Bride Part II (1995)","6":"Heat (1995)","7":"Sabrina (1995)","8":"Tom and Huck (1995)","9":"Sudden Death (1995)"},"2":{"0":"genres","1":"Adventure|Animation|Children|Comedy|Fantasy","2":"Adventure|Children|Fantasy","3":"Comedy|Romance","4":"Comedy|Drama|Romance","5":"Comedy","6":"Action|Crime|Thriller","7":"Comedy|Romance","8":"Adventure|Children","9":"Action"}}
======
/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/io/json.py:70: UserWarning: Using CPU via Pandas to write JSON dataset, this may be GPU accelerated in the future
  "Using CPU via Pandas to write JSON dataset, this may "
None
>>> 
```

to_json function in cudf `https://github.com/rapidsai/cudf/blob/branch-0.13/python/cudf/cudf/io/json.py`
```
@ioutils.doc_to_json()
def to_json(cudf_val, path_or_buf=None, *args, **kwargs):
    """{docstring}"""

    warnings.warn(
        "Using CPU via Pandas to write JSON dataset, this may "
        "be GPU accelerated in the future"
    )
    pd_value = cudf_val.to_pandas()
    pd.io.json.to_json(path_or_buf, pd_value, *args, **kwargs)
```